### PR TITLE
fix: correct hostname for gin sample and add compose file to run anywhere to test

### LINKED ISSUE #107: bug: lookup 0.0.0: no such host
### Describe the bug
`[GIN-debug] [ERROR] listen tcp: lookup 0.0.0: no such host`

### Expected behavior
Gin to start on 0.0.0.0 with port 8080

### How to reproduce?
Run code in docker

```yaml
services:
  go:
    build:
      context: .
      dockerfile_inline: |
        FROM golang:1.22-alpine
        RUN apk add git
        WORKDIR /app
        RUN git clone https://github.com/logto-io/go.git
        WORKDIR /app/go/gin-sample
        RUN go mod download
        CMD go run .
    ports:
      - 8080:8080
```

### PR DESCRIPTION
**Pull request by flohoss**
## Summary
Fix a zero missing in host and add compose file to run anywhere to test.

Closes #107

### CODE REVIEW

File: gin-sample/main.go [hunk 1]
```diff
@@ -15,10 +16,10 @@ var (
 
 func main() {
 	logtoConfig := &client.LogtoConfig{
-		Endpoint:  "<your-logto-endpoint>", // E.g. "http://localhost:3001"
-		AppId:     "<your-app-id>",
-		AppSecret: "<your-app-secret>",
-		Resources: []string{"your-resource"}, // E.g. "https://default.logto.app/api"
+		Endpoint:  os.Getenv("ENDPOINT"),
```
Comment (xiaoyijun) [inline gin-sample/main.go:19]: The example config needs to be reserved, since some users will confuse about it.

File: gin-sample/main.go [hunk 2]
```diff
@@ -15,10 +16,10 @@ var (
...
+		AppId:     os.Getenv("APP_ID"),
+		AppSecret: os.Getenv("APP_SECRET"),
+		Resources: []string{os.Getenv("RESOURCES")},
```
Comment (xiaoyijun) [inline gin-sample/main.go:22]: Ditto.

File: gin-sample/main.go [hunk 3]
```diff
@@ -51,7 +52,7 @@ func main() {
 	router.GET("/sign-in", func(ctx *gin.Context) {
 		session := sessions.Default(ctx)
 		logtoClient := client.NewLogtoClient(logtoConfig, &SessionStorage{session: session})
-		signInUri, err := logtoClient.SignIn("http://localhost:8080/sign-in-callback")
+		signInUri, err := logtoClient.SignIn(os.Getenv("CALLBACK_URI"))
```
Comment (xiaoyijun) [inline gin-sample/main.go:55]:
```suggestion
		signInUri, err := logtoClient.SignIn(os.Getenv("REDIRECT_URI"))
```
To keep align with the field in the console:
image

File: gin-sample/main.go [hunk 4]
```diff
@@ -89,7 +90,7 @@ func main() {
 		session := sessions.Default(ctx)
 		logtoClient := client.NewLogtoClient(logtoConfig, &SessionStorage{session: session})
 
-		signOutUri, signOutErr := logtoClient.SignOut("http://localhost:8080")
+		signOutUri, signOutErr := logtoClient.SignOut(os.Getenv("POST_LOGOUT_REDIRECT_URI"))
```
Comment (xiaoyijun) [inline gin-sample/main.go:93]:
```suggestion
		signOutUri, signOutErr := logtoClient.SignOut(os.Getenv("POST_SIGN_OUT_REDIRECT_URI"))
```
Ditto.

**Review by xiaoyijun — APPROVED** [review]
Overall LGTM.

**Comment by xiaoyijun** [comment]
Nice job @flohoss , but some updates needed. I'll merge this PR once you've updated related codes.
Thanks again 🫶

File: gin-sample/compose.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        RUN git clone https://github.com/logto-io/go.git
+        WORKDIR /app/go/gin-sample
+        RUN go mod download
+
+        ENV ENDPOINT=<your-logto-endpoint>
+        ENV APP_ID=<your-app-id>
+        ENV APP_SECRET=<your-app-secret>
+        ENV RESOURCES=<your-resources>
+        ENV CALLBACK_URI=<your-callback-uri>
+        ENV POST_LOGOUT_REDIRECT_URI=<your-post-logout-redirect-uri>
```
Comment (xiaoyijun) [inline gin-sample/compose.yml:18]: These key names should be updated accordingly.

File: gin-sample/main.go (hunk 1 above)
Comment (flohoss) [reply gin-sample/main.go:21]: Moved examples into a .env.example file for easy use

File: gin-sample/main.go (hunk 3 above)
Comment (flohoss) [reply gin-sample/main.go]: Done.

**Pull request merged**